src/test.js

```diff
@@ -656,20 +657,23 @@ export function only( testName, callback ) {
 
 // Put a hold on processing and return a function that will release it.
 export function internalStop( test ) {
-	var released = false;
-
 	test.semaphore += 1;
 	config.blocking = true;
 
 	// Set a recovery timeout, if so configured.
-	if ( config.testTimeout && defined.setTimeout ) {
+	const timeoutDuration = test.timeout || config.testTimeout;
+	if ( timeoutDuration && defined.setTimeout ) {
```

Comment:
Per the above, this should be more explicit. There's also a question of whether the global value is a fallback or a _maximum_. I'm personally inclined towards the latter, e.g.
```js
const specificTimeout = typeof test.timeout === "number" ? test.timeout : Infinity;
const globalTimeout = typeof config.testTimeout === "number" ? config.testTimeout : Infinity;
const resolvedTimeout = Math.min( explicitTimeout, globalTimeout );
if ( resolvedTimeout < Infinity && defined.setTimeout ) {
	…
}
```